src/plugins/discover/public/application/apps/main/components/layout/discover_layout.tsx

```diff
@@ -244,18 +289,16 @@ export function Discover({
           <EuiFlexGroup className="dscPageBody__contents" gutterSize="none">
             <EuiFlexItem grow={false}>
               <SidebarMemoized
-                config={config}
                 columns={columns}
-                fieldCounts={fieldCounts}
-                hits={rows}
+                fieldCounts={fetchState.fieldCounts!}
```

Comment:
Are these safe? It seems like they are accessed in a way that could cause exceptions if these variables are ever undefined. Maybe there is a good reason, but as this relationship is spanning multiple files IMHO it makes sense to default to an empty object of some sort here to not introduce a bug in the future.

Reply:
I've improved the typings there. So there's now a type for components local state:

```TypeScript
interface DiscoverLayoutFetchState extends SavedSearchDataMessage {
  state: FetchStatus;
  fetchCounter: number;
  fieldCounts: Record<string, number>;
  rows: ElasticSearchHit[];
}
```

this gets updated by the the saved search data observable . Since fieldCounts are set in the initial state, it can't be undefined